src/app/operators/dialogs/box-plot-operator/box-plot-operator.component.ts

```diff
@@ -10,33 +10,38 @@ import {DataType, DataTypes} from '../../datatype.model';
 import {NumericPipe} from '../scatter-plot-operator/scatter-plot-operator.pipe';
 import {WaveValidators} from '../../../util/form.validators';
 import {BoxPlotType} from '../../types/boxplot-type.model';
+import {BehaviorSubject} from 'rxjs';
 
 @Component({
     selector: 'wave-box-plot-operator',
     templateUrl: './box-plot-operator.component.html',
     styleUrls: ['./box-plot-operator.component.scss'],
     changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class BoxPlotComponent implements OnInit, AfterViewInit {
+export class BoxPlotComponent implements ControlValueAccessor, OnInit, AfterViewInit {
 
     form: FormGroup;
 
     pointLayers: Array<Layer<Symbology>>;
 
     ResultTypes = ResultTypes;
-    NumericPipe = NumericPipe;
     DataTypes = DataTypes;
 
+    onTouched: () => void;
+    onChange: (_: Array<{attr: string}>) => void = undefined;
+
+    selectedLayers = new BehaviorSubject<Array<{attr: string}>>([]);
+
     max = 5;
 
     constructor(private formBuilder: FormBuilder,
-                private projectService: ProjectService) {
+                private projectService: ProjectService,
+                private _changeDetectorRef: ChangeDetectorRef) {
```

Comment:
If it is private there is actually no need to use an underscore. This is more likely used in languages where you do not have the differentiation between public and private fields (like e.g. Python) or when you implement an interface method and do not use an input variable (e.g. Rust). But this is just a stylistic issue.